poc_iot_injector/src/server.rs

```diff
@@ -136,17 +162,23 @@ async fn submit_txns(
 }
 
 async fn handle_txn_submission(
-    msg: prost::bytes::BytesMut,
-    shared_key: Arc<Keypair>,
+    txn_details: TxnDetails,
     txn_service: &mut TransactionService,
-    before_ts: i64,
-) -> Result<BlockchainTxn> {
-    let (txn, hash, hash_b64_url) = handle_report_msg(msg, shared_key, before_ts)?;
-    if txn_service.submit(txn.clone(), &hash).await.is_ok() {
-        tracing::debug!("txn submitted successfully, hash: {:?}", hash_b64_url);
-        Ok(txn)
+) -> Result {
+    if txn_service
+        .submit(txn_details.txn, &txn_details.hash)
+        .await
+        .is_ok()
+    {
+        tracing::debug!(
+            "txn submitted successfully, hash: {:?}",
+            txn_details.hash_b64_url
+        );
     } else {
-        tracing::warn!("txn submission failed!, hash: {:?}", hash_b64_url);
-        Err(Error::TxnSubmission(hash_b64_url))
+        tracing::warn!(
+            "txn submission failed!, hash: {:?}",
+            txn_details.hash_b64_url
```

Comment:
i think i see @andymck 's point here (didn't read ahead); are we sure we want to take out this submission error here? when the `do_submission` boolean is `false` and we're just writing out the proposed txn without submitting it we get the validation of the process without submitting or triggering any rewards but when we deliberately set that boolean to true wouldn't we want some sort of guard in place here in case something goes wrong with that submission?

also with the unconditional `Ok(())` below the `is_ok()` check on the `if handle_txn_submission` above seems unnecessary; if this is just a temporary setup and `handle_txn_submission` meant to potentially return Ok _and_ Err responses to the caller can we get a todo/comment here to note for later address?

Reply:
Yup, that's fair. I'll bubble up the error if the txn submission fails so the whole `submi_txn` should fail if the node is down/unreachable. So when the next tick happens the `last_poc_submission_ts` wouldn't have moved forward and we'd still process the txn which went unsubmitted.

Reply:
@andymck @jeffgrunewald Updated [13d7cf0](https://github.com/helium/oracles/pull/187/commits/13d7cf0457a382784e702f087f403e87806add9c).

So now we'll exit early if there's any error in `handle_report_msg` or `handle_txn_submission`, which also means that `submit_txn` itself would exit thus not updating the `last_poc_submission_ts`, so at the next we should pick up the txns we were not able to process.

Note that we would run into duplicate submissions from the previous time range, but that's something the chain should be able to handle and drop.